controllers/clusterversion_controller.go

```diff
@@ -124,47 +122,181 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	var err error
 	c.ctx = ctx
 	c.log = log.FromContext(ctx, "ClusterVersion", req)
-	c.log.Info("Reconciling ClusterVersion")
 
-	if err := c.ensureConsolePlugin(); err != nil {
-		c.log.Error(err, "unable to deploy client console")
+	c.log.Info("Loading input resources")
+
+	// Load the cluster version
+	c.clusterVersion.Name = req.Name
+	c.clusterVersion.Namespace = req.Namespace
+	if err := c.get(&c.clusterVersion); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Load the console plugin
+	c.consoleDeployment.Name = console.DeploymentName
+	c.consoleDeployment.Namespace = c.OperatorNamespace
+	if err := c.get(&c.consoleDeployment); err != nil {
+		c.log.Error(err, "failed to get the deployment for the console")
+		return ctrl.Result{}, err
+	}
+
+	// Load the operator configmap
+	c.operatorConfigMap.Name = operatorConfigMapName
+	c.operatorConfigMap.Namespace = c.OperatorNamespace
+	if err := c.get(&c.operatorConfigMap); err != nil && !k8serrors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	}
 
-	instance := configv1.ClusterVersion{}
-	if err = c.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
+	c.log.Info("Starting reconciliation")
+	if result, err := c.reconcilPhases(); err == nil {
+		c.log.Error(err, "An error was encountered during reconciliation")
 		return ctrl.Result{}, err
+	} else {
+		c.log.Info("Reconciliation completed successfully")
+		return result, nil
 	}
+}
+
+func (c *ClusterVersionReconciler) reconcilPhases() (ctrl.Result, error) {
 
-	if err := csi.InitializeSidecars(c.log, instance.Status.Desired.Version); err != nil {
+	// Reconcile console related resources
+	if err := c.reconcileNginxConfigMap(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsoleService(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsolePlugin(); err != nil {
+		c.log.Error(err, "unable to deploy client console")
+		return ctrl.Result{}, err
+	}
+
+	// Reconcile CSI related resources
+	if err := csi.InitializeSidecars(c.log, c.clusterVersion.Status.Desired.Version); err != nil {
 		c.log.Error(err, "unable to initialize sidecars")
```

Comment:
only these two logs seem odd, other methods are logging and returning an error, maybe follow the same?